EzMap.Api/Services/ElasticSearchService.cs

```diff
@@ -0,0 +1,89 @@
+using Elasticsearch.Net;
+using Nest;
+
+namespace EzMap.Api.Services;
+
+public interface IElasticSearchService
+{
+    Task CreateIndexIfNotExists(string indexName);
+    Task<bool> AddOrUpdateBulk(IEnumerable<object> documents);
+    Task<bool> AddOrUpdate(object document);
+    Task<object> Get(string key);
+    Task<List<object>?> GetAll();
+    Task<List<object>?> Query(QueryContainer predicate);
+    Task<bool> Remove(string key);
+    Task<long> RemoveAll();
+}
+
+public class ElasticSearchService : IElasticSearchService
+{
+    private string _indexName { get; set; }
+    private readonly IElasticClient _client;
+
+    public ElasticSearchService(IElasticClient client)
+    {
+        _indexName = "thanh1";
+        _client = client;
+    }
+
+    public ElasticSearchService Index(string indexName)
```

Comment:
Is it index, or is it CreateIndex?

Reply:
it is CreateIndex if it is not existed, refactoring this